src/main/java/com/petros/bibernate/dao/EntityPersister.java

```diff
@@ -194,7 +198,8 @@ private static void throwExceptionIfRowsAffectedNotOne(int rowsAffected, String
         }
     }
 
-    private static <T> PreparedStatement prepareInsertStatement(T entity, Connection connection) throws SQLException {
+    private <T> PreparedStatement prepareInsertStatement(T entity, Connection connection) throws SQLException {
+        validateJoinColumnsOnInsertOperation(entity);
```

Comment:
I think we should remove this validation:

- Firstly, all tests pass if we comment out this line, and this check will be performed on the database side.
- Secondly, this validation will perform an additional Select query to the database, which is undesirable because it will affect performance.

Reply:
Agree. Removed this validation.